base_layer/contacts/src/contacts_service/service.rs

```diff
@@ -582,8 +582,10 @@ where T: ContactsBackend + 'static
         message: Message,
         source_public_key: CommsPublicKey,
     ) -> Result<(), ContactsServiceError> {
+        if source_public_key != *message.from_address.public_key() {
+            return Err(ContactsServiceError::MessageSourceDoesNotMatchOrigin);
+        }
```

Comment:
Under the new address scheme, we can match the comms public key to the public view key. 
However, we can't validate the public spend key which makes up the second part of their complete address.

Is there any way a sender could use this as an attack vector by populating the `from_address` field with an address that contains a matching view key, and a non-matching spend key? I can't think of a way it would be useful, but mentioning it just to make sure.

Reply:
Comms_key == spend key, 
This key is super important to keep safe, as losing this you lose your utxos. 

But on the view key, its purely there for you to advertise your full address to your message contact as sender. From the contact service, the view key has no purpose or use. But if the sender where to send you a "fake" view key, they might not receive one-sided transactions that you where to send to him. So the sender has no incentive to lie about this and the receiver does not use the key here, so there is no attack vector here.